tests/FakeItEasy.Tests/Core/FakeManagerTests.cs

```diff
@@ -39,7 +39,7 @@ public void Event_listeners_that_are_removed_should_not_be_invoked_when_event_is
         public void Method_call_should_return_default_value_when_there_is_no_matching_interception_and_return_type_is_value_type()
         {
             var fake = this.CreateFakeManager<IFoo>();
-            var result = ((IFoo)fake.Object).Baz();
+            var result = ((IFoo)fake.Object!).Baz();

```

Comment:
Here, it looks like the fake *could* have been collected, so `fake.Object` could return null. However, in `CreateFakeManager`, we call `MakeSureThatWeakReferenceDoesNotGetCollected`, which adds a strong reference to the fake in a list. So `fake.Object` won't return null.